imagetest/test_suites/disk/disk_resize_test.go

```diff
@@ -0,0 +1,68 @@
+package disk
+
+import (
+	"flag"
+	"fmt"
+	"math"
+	"os"
+	"os/exec"
+	"strconv"
+	"strings"
+	"testing"
+)
+
+const (
+	defaultDiskSize = 20
+	gb              = 1024.0 * 1024.0
+	markerFile      = "/boot-marker"
+)
+
+var (
+	runtest = flag.Bool("runtest", false, "really run the test")
+)
+
+func TestMain(m *testing.M) {
+	flag.Parse()
+	if *runtest {
+		os.Exit(m.Run())
+	} else {
+		os.Exit(0)
+	}
+}
+
+// TestDiskResize Validate the filesystem is resized on reboot after a disk resize.
+func TestDiskResize(t *testing.T) {
+	_, err := os.Stat(markerFile)
+	if os.IsNotExist(err) {
```

Comment:
in this test i think you can just check if the disk is the correct post-resize size in both boots. it'll upload a failure result on the first boot, and (in theory) a success result on the second boot. then you can avoid this first-boot/second-boot switch